File: packages/java/src/main/java/io/axene/mailer/Attachment.java
package io.axene.mailer;

import java.util.LinkedHashMap;
import java.util.Map;

/** A file attachment. {@code contentBase64} is the base64-encoded content. */
public final class Attachment {
    public final String filename;
    public final String contentBase64;
    public final String contentType;

    public Attachment(String filename, String contentBase64) { this(filename, contentBase64, null); }
    public Attachment(String filename, String contentBase64, String contentType) {
        this.filename = filename; this.contentBase64 = contentBase64; this.contentType = contentType;
    }

    Map<String, Object> toMap() {
        Map<String, Object> m = new LinkedHashMap<>();
        m.put("filename", filename);
        m.put("content_base64", contentBase64);
        if (contentType != null) m.put("content_type", contentType);
        return m;
    }
}

File: packages/dotnet/src/Models/Attachment.cs
using System.Text.Json.Serialization;

namespace Axene.Mailer
{
    /// <summary>A file attachment.</summary>
    public sealed class Attachment
    {
        /// <summary>The file name shown to the recipient.</summary>
        [JsonPropertyName("filename")] public string Filename { get; set; } = "";

        /// <summary>Base64-encoded file content.</summary>
        [JsonPropertyName("content_base64")] public string ContentBase64 { get; set; } = "";

        /// <summary>Optional MIME type (inferred from the filename if omitted).</summary>
        [JsonPropertyName("content_type")] public string? ContentType { get; set; }
    }
}
